### PR TITLE
feat(pie-storybook): DSW-1591 add number input functionality

### PR DESCRIPTION
**Pull request by jamieomaguire**
## Author Checklist (complete before requesting a review)
- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests
- [x] If it is a component change, I have reviewed the Storybook preview

### CODE REVIEW

File: packages/components/pie-input/src/input.scss [hunk 1]
```diff
@@ -1 +1,11 @@
 @use '@justeattakeaway/pie-css/scss' as p;
+
+input[type="number"]:not([step])::-webkit-inner-spin-button,
```
Comment (jamieomaguire) [inline packages/components/pie-input/src/input.scss:3]: This is placeholder styling which will be replaced in the upcoming styling ticket. Please disregard it.

Comment (xander-marjoram) [reply packages/components/pie-input/src/input.scss:3]: To-do comment with ticket reference?

File: packages/components/pie-input/test/component/pie-input.spec.ts [hunk 2]
```diff
@@ -566,7 +566,7 @@ test.describe('PieInput - Component tests', () => {
                 expect(assistiveText).not.toBeVisible();
             });
 
-            test('should render the assistive text component if the prop is provided with the default variant if "status" is not provided', async ({ mount, page }) => {
+            test('should render the default variant if no status provided', async ({ mount, page }) => {
```
Comment (xander-marjoram) [inline packages/components/pie-input/test/component/pie-input.spec.ts:569]: Is the "Default text" inherent to the default variant? Not sure I fully understand this text

File: packages/tools/pie-wrapper-react/custom-elements.json [hunk 3]
```diff
@@ -1,43 +0,0 @@
-{
-    "schemaVersion": "1.0.0",
-    "readme": "",
-    "modules": [
-        {
-            "kind": "javascript-module",
-            "path": "pie-wrapper-react",
```
Comment (xander-marjoram) [inline packages/tools/pie-wrapper-react/custom-elements.json:7]: I'm guessing this file isn't required? There isn't much here so fair enough if so

Comment (jamieomaguire) [reply packages/tools/pie-wrapper-react/custom-elements.json:7]: This was added to the .gitignore a while ago but lingered in the git cache. This correctly deletes the file from source control :)

**Review by siggerzz — APPROVED** [review]

**Review by fernandofranca — DISMISSED** [review]

**Review by fernandofranca — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]

File: packages/components/pie-input/src/index.ts [hunk 4]
```diff
@@ -154,8 +163,11 @@ export class PieInput extends FormControlMixin(RtlMixin(LitElement)) implements
             inputmode,
             maxlength,
             minlength,
+            min,
+            max,
             name,
             pattern,
+            step,
```
Comment (ashleynolan) [inline packages/components/pie-input/src/index.ts:170]: Sorry for the delay commenting on this (was just having a nose back through the last weeks worth I have built up!), but if I didn't NIT on `step` not being in alphabetical order like the rest of the attributes, I couldn't live with myself 😆  (`max` as well though too)

(I also can't believe @xander-marjoram let this one through 🤣 )

Comment (ashleynolan) [reply packages/components/pie-input/src/index.ts:170]: Also, everything else here looks great btw 💯 🚀

Comment (jamieomaguire) [reply packages/components/pie-input/src/index.ts:170]: I did it to hurt you @ashleynolan

**Pull request merged**